com.mysql.jdbc.Driver is deprecated

Autodetection of the database driver is defaulting to the deprecated version of the MySQL driver:

https://github.com/micronaut-projects/micronaut-sql/blob/master/jdbc/src/main/java/io/micronaut/jdbc/JdbcDatabaseManager.java#L42

This should be replaced with `com.mysql.cj.jdbc.Driver`

This also means by default things are broken in GraalVM because the feature is looking for the new driver:

https://github.com/micronaut-projects/micronaut-sql/blob/master/jdbc/src/main/java/io/micronaut/jdbc/nativeimage/JdbcFeature.java#L55

So GraalVM fails with:

```
Caused by: java.lang.RuntimeException: Failed to load driver class com.mysql.jdbc.Driver in either of HikariConfig class loader or Thread context classloader
        at com.zaxxer.hikari.HikariConfig.setDriverClassName(HikariConfig.java:486)
        at io.micronaut.configuration.jdbc.hikari.DatasourceConfiguration.postConstruct(DatasourceConfiguration.java:70)
        at io.micronaut.configuration.jdbc.hikari.$DatasourceConfigurationDefinition.initialize(Unknown Source)
```

And actually the new driver is not working on GraalVM:

```
Caused by: java.lang.ClassCastException: com.mysql.cj.exceptions.CJException cannot be cast to com.mysql.cj.exceptions.UnableToConnectException
        at com.mysql.cj.jdbc.ConnectionImpl.createNewIO(ConnectionImpl.java:833)
        at com.mysql.cj.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:456)
        ... 26 more
```

--- a/jdbc/src/main/java/io/micronaut/jdbc/JdbcDatabaseManager.java
+++ b/jdbc/src/main/java/io/micronaut/jdbc/JdbcDatabaseManager.java
@@ -39,7 +39,7 @@ public class JdbcDatabaseManager {
         databases.add(new EmbeddedJdbcDatabase("org.apache.derby.jdbc.EmbeddedDriver", "SELECT 1 FROM SYSIBM.SYSDUMMY1", new String[]{"derby"}, "jdbc:derby:memory:%s;create=true"));
         databases.add(new EmbeddedJdbcDatabase("org.hsqldb.jdbc.JDBCDriver", "select 1 from INFORMATION_SCHEMA.SYSTEM_USERS", new String[]{"hsqldb"}, "jdbc:hsqldb:mem:%s"));
 
-        databases.add(new JdbcDatabase("com.mysql.jdbc.Driver", "mysql"));
+        databases.add(new JdbcDatabase("com.mysql.cj.jdbc.Driver", "mysql"));
         databases.add(new JdbcDatabase("oracle.jdbc.OracleDriver", "SELECT 1 FROM DUAL", new String[]{"oracle"}));
         databases.add(new JdbcDatabase("org.postgresql.Driver", "postgresql"));
         databases.add(new JdbcDatabase("com.microsoft.sqlserver.jdbc.SQLServerDriver", "sqlserver"));

--- a/jdbc/src/main/java/io/micronaut/jdbc/nativeimage/JdbcFeature.java
+++ b/jdbc/src/main/java/io/micronaut/jdbc/nativeimage/JdbcFeature.java
@@ -18,6 +18,7 @@ package io.micronaut.jdbc.nativeimage;
 import com.oracle.svm.core.annotate.AutomaticFeature;
 import com.oracle.svm.core.configure.ResourcesRegistry;
 import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.graal.AutomaticFeatureUtils;
 import org.graalvm.nativeimage.hosted.Feature;
 
 import java.util.Arrays;
@@ -219,6 +220,36 @@ final class JdbcFeature implements Feature {
     private void handleMySql(BeforeAnalysisAccess access) {
         Class<?> mysqlDriver = access.findClassByName(MYSQL_DRIVER);
         if (mysqlDriver != null) {
+            Arrays.asList(
+                    "com.mysql.cj.exceptions.AssertionFailedException",
+                    "com.mysql.cj.exceptions.CJCommunicationsException",
+                    "com.mysql.cj.exceptions.CJConnectionFeatureNotAvailableException",
+                    "com.mysql.cj.exceptions.CJException",
+                    "com.mysql.cj.exceptions.CJOperationNotSupportedException",
+                    "com.mysql.cj.exceptions.CJPacketTooBigException",
+                    "com.mysql.cj.exceptions.CJTimeoutException",
+                    "com.mysql.cj.exceptions.ClosedOnExpiredPasswordException",
+                    "com.mysql.cj.exceptions.ConnectionIsClosedException",
+                    "com.mysql.cj.exceptions.DataConversionException",
+                    "com.mysql.cj.exceptions.DataReadException",
+                    "com.mysql.cj.exceptions.DataTruncationException",
+                    "com.mysql.cj.exceptions.FeatureNotAvailableException",
+                    "com.mysql.cj.exceptions.InvalidConnectionAttributeException",
+                    "com.mysql.cj.exceptions.MysqlErrorNumbers",
+                    "com.mysql.cj.exceptions.NumberOutOfRange",
+                    "com.mysql.cj.exceptions.OperationCancelledException",
+                    "com.mysql.cj.exceptions.PasswordExpiredException",
+                    "com.mysql.cj.exceptions.PropertyNotModifiableException",
+                    "com.mysql.cj.exceptions.RSAException",
+                    "com.mysql.cj.exceptions.SSLParamsException",
+                    "com.mysql.cj.exceptions.StatementIsClosedException",
+                    "com.mysql.cj.exceptions.UnableToConnectException",
+                    "com.mysql.cj.exceptions.UnsupportedConnectionStringException",
+                    "com.mysql.cj.exceptions.WrongArgumentExceptio"
+            ).forEach(name -> {
+                AutomaticFeatureUtils.registerClassForRuntimeReflection(access, name);
+                AutomaticFeatureUtils.registerConstructorsForRuntimeReflection(access, name);
+            });
             registerFieldsAndMethodsWithReflectiveAccess(access, MYSQL_DRIVER);
 
             registerAllForRuntimeReflection(access, "com.mysql.cj.log.StandardLogger");

